### PR TITLE
Nas work

### PR DESCRIPTION
**Pull request by samiamseid**
Created tool to return list of sessions in NAS that have already been released

### CODE REVIEW

File: meso_tools/NAS_tools.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    lines =[]
+    for line in credentials[1:]:
+        line_split = line.split('=')
+        for items in line_split:
+            item = items.split('\n')[0]
+            lines.append(item)
+    keys = lines[::2]
+    values = lines[1::2]
+    res = {keys[i]: values[i] for i in range(len(keys))}
+    return(res)
```
Comment (nataliaorlova) [inline meso_tools/NAS_tools.py:32]: no need for parenthesis after return

File: meso_tools/NAS_tools.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """Compares the output of a requests query against a list of released sessions, 
+    and returns the filepath for each session in the NAS storage that matches the ID of a released session
+
+    Args:
+        release_list (str): Relative string filepath to a .csv file containing information about released ophys sessions. 
+        This data is the result of an AllenSDK query and not currently included in this code
+        query_response (requests.models.Response): Response from the NAS storage containing list of folders and filepaths.  
+        This data is the output of the nas_query function
+    """    
+    import pandas as pd
```
Comment (nataliaorlova) [inline meso_tools/NAS_tools.py:88]: do this in the import section at the top please

**Review by nataliaorlova — CHANGES_REQUESTED** [review]
Generally, I'd make this a class... "nas_api" maybe, where instantiating a class (__init__ method) makes connection and takes in credential file, and other methods can do things with data, for instance, query folder list, query available space, session list, remove folder, create folder, create duplicate, etc. Similar to lims_api class in io_utils.py

File: meso_tools/NAS_tools.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        print('NAS name not valid')
+        released_backups = []
+        released_backups2 = []
+    # logout
+
+    requests.get("http://"+str(ip)+"/webapi/auth.cgi?api=SYNO.API.Auth&version=6&method=logout&session=FileStation")
+    return(released_backups, released_backups2)
+            
+def release_check(release_list: str, query_response: requests.models.Response):
+    """Compares the output of a requests query against a list of released sessions,
```
Comment (nataliaorlova) [inline meso_tools/NAS_tools.py:79]: Let's use Numpy format for docstrings - this is set is autodocstring plugin settings

File: meso_tools/NAS_tools.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        line_split = line.split('=')
+        for items in line_split:
+            item = items.split('\n')[0]
+            lines.append(item)
+    keys = lines[::2]
+    values = lines[1::2]
+    res = {keys[i]: values[i] for i in range(len(keys))}
+    return(res)
+
+def nas_query(credentials: dict, nas_name: str, release_list: str):
```
Comment (nataliaorlova) [inline meso_tools/NAS_tools.py:34]: add output type using 
def function_name(input1 : str, input2 : int) -> int:
   body of the function
   return output (integer)

**Review by nataliaorlova — CHANGES_REQUESTED** [review]

File: meso_tools/NAS_tools.py (hunk 1 above)
Comment (samiamseid) [reply meso_tools/NAS_tools.py]: done

File: meso_tools/NAS_tools.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    Expected format for txt doc:
+    
+    # Device access credentials
+    SYNOLOGY_IP=xx.xxx.xx.xx:xxxx
+    SYNOLOGY_PORT=xxxx
+    SYNOLOGY_USERNAME=x
+    SYNOLOGY_PASSWORD=x
+
+    Args:
+        cred_path (str): relative string path of txt document containing NAS credentials
```
[the remark this answers is not in the document]
Comment (samiamseid) [reply meso_tools/NAS_tools.py]: done

File: meso_tools/NAS_tools.py (hunk 2 above)
Comment (samiamseid) [reply meso_tools/NAS_tools.py]: done

File: meso_tools/NAS_tools.py (hunk 4 above)
Comment (samiamseid) [reply meso_tools/NAS_tools.py]: done

File: meso_tools/NAS_tools.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            delete_list = self.released_backups  
+        task_dict = {}
+        for output in delete_list:
+            print('deleting '+str(output))
+            #start deletion
+            response = requests.get("http://"+self.ip+"/webapi/entry.cgi?api=SYNO.FileStation.Delete&version=2&method=start&path=%22%2F"+str(output[1:])+"%22&_sid="+self.sid)
+            #save task id for the current deletion jobs
+            task_dict[output.split('/')[-1]]=response.json()['data']['taskid']
+        #store task ID's for later in case we need to stop a job
+        self.task_ids = task_dict
```
Comment (nataliaorlova) [inline meso_tools/NAS_tools.py:147]: add self.task_id to init

**Review by nataliaorlova — CHANGES_REQUESTED** [review]

**Review by nataliaorlova — APPROVED** [review]

**Pull request merged**